### PR TITLE
uhv: add http-parser permissive parsing mode for h1 codec

### LINKED ISSUE #19750: uhv: http-parser: add option to skip header validation
As part of the unified header validation component, the http-parser library will be modified to add an option to disable header validation on a per-request basis. The intent is that header validation will be performed by the UHV and not by each codec / library. 

http-parser currently controls how header validation is performed on the compile-time [`HTTP_PARSER_STRICT`](https://github.com/nodejs/http-parser/blob/main/http_parser.h#L50) preprocessor definition, which is enabled by default. This issue will change http-parser to make this configurable at runtime via a new option.

This is blocked until #19749 is completed.

### PR DESCRIPTION
**Pull request by ameily**
Commit Message: uhv: add http-parser permissive parsing mode for h1 codec

Additional Description: Adds a new setting to http-parser, `permissive_parsing` which enables permissive parsing mode (disabled by default). Strict vs permissive parsing was previously controlled by a compile time definition, `HTTP_PARSER_STRICT`, and my work was primarily making this a runtime flag on a per-message basis. I brought in the http-parser unit tests to make sure that nothing broke in the conversion. Then, based on the PR for H2, #20381, this underlying setting is controlled with the `CodecHeaderValidationMode` enumeration in the H1 codec and legacy parser. For now, all code that creates an H1 codec or parser are hardcoded to set the validation mode to `Enabled` to preserve the current behavior.

Risk Level: Low

Testing:

Docs Changes:

Release Notes:

Platform Specific Features:

Fixes #19750

### CODE REVIEW

**Comment by ameily** [comment]
@yanavlasov Can you take a look at this when you get a chance (assuming CI passes)?

**Comment by phlax** [comment]
copying out CI advice

```console

  Please fix your editor to ensure:

      - no trailing whitespace
      - no preceding mixed tabs/spaces
      - all files end with a newline
```

**Comment by adisuissa** [comment]
@yanavlasov can you PTAL?

File: source/common/http/http1/BUILD [hunk 1]
```diff
@@ -108,11 +108,21 @@ envoy_cc_library(
 
 envoy_cc_library(
     name = "legacy_parser_lib",
-    srcs = ["legacy_parser_impl.cc"],
-    hdrs = ["legacy_parser_impl.h"],
-    external_deps = ["http_parser"],
+    srcs = [
+        "legacy_parser_impl.cc",
+        "legacy_permissive_parser_impl.cc",
```
Comment (yanavlasov) [inline source/common/http/http1/BUILD:113]: In general with bazel the preference is to create fine grained targets. In this case the legacy_permissive_parser_impl.cc will be in its own library

**Review by yanavlasov — CHANGES_REQUESTED** [review]
Thanks this looks great. Just one small nit

/wait

File: source/common/http/http1/BUILD (hunk 1 above)
Comment (ameily) [reply source/common/http/http1/BUILD]: Done.

**Comment by ameily** [comment]
The tests that are failing in CI are working fine locally and were previously running successfully before my last commit, which shouldn't have affected them.

**Comment by alyssawilk** [comment]
I think we can get away with build options and testing on the compile time build - I'll let Yan touch base with you on details but please ping back after you sync with him if you have follow up questions?

**Comment by ameily** [comment]
I'm closing this in favor of a new solution that builds upon the `uhv_enabled` build flag.

**Pull request closed without merging**